labs/data_encoding_health.py

```diff
@@ -0,0 +1,116 @@
+
+import pandas as pd
+from helpers.dslabs_functions import get_variable_types, encode_cyclic_variables, dummify
+import numpy as np
+
+
+def calulate_mean_age(age_categories):
+    mean_ages = []
+    for value in age_categories:
+        split = value.split()
+        age1, age2 = int(split[1]), split[3]
+        if age2.isdigit():
+            mean_ages.append((age1+int(age2))/2)
+        else:
+            mean_ages.append(85)
+    return mean_ages
+
+def yes_no_mapping(answer):
+    if pd.isna(answer):
+        return np.nan
+    elif answer.startswith('Yes'):
+        return 1
+    elif answer.startswith('No'):
+        return 0
+    else:
+        return np.nan  # or a specific code for answers that do not start with Yes/No
+
+
+def parse_location_file(file_path):
+    location_mapping = {}
+
+    with open(file_path, 'r') as file:
+        next(file)  # Skip the header line
+        for line in file:
+            # Remove quotation marks and split the line
+            parts = line.strip().replace('"', '').split(':')
+            state = parts[0].strip()
+            coords = parts[1].strip().strip('[]').split(',')
+            longitude = float(coords[0].strip())
+            latitude = float(coords[1].strip())
+
+            location_mapping[state] = {'Latitude': latitude, 'Longitude': longitude}
+
+    return location_mapping
+
+def add_coordinates(df, mapping):
+    # Extract latitude and longitude from the mapping
+    df['Latitude'] = df['State'].map(lambda x: mapping[x]['Latitude'] if x in mapping else None)
+    df['Longitude'] = df['State'].map(lambda x: mapping[x]['Longitude'] if x in mapping else None)
+    return df
+
+def encode_symbolic(df, encoding):
+    
+    df = pd.get_dummies(df, columns=['RaceEthnicityCategory'], dummy_na=False)
+    transform_bools(df, "RaceEthnicityCategory")
+    file_path = "../datasets/State, LON, LAT.txt"
+    state_encoding = parse_location_file(file_path)
+    df = add_coordinates(df, state_encoding)
+    df = df.drop(columns=["State"])
+    df = df.replace(encoding, inplace=False)
+    return df
+
+def transform_bools(df, keyword):
+    for col in df.columns:
+        if col.startswith(keyword):
+            df[col] = df[col].astype(int)
+    return df
+
+def encode_health():
+    df = pd.read_csv('../datasets/class_pos_covid.csv')
+
+    #Binaries
+    yes_no: dict[str, int] = {"no": 0, "No": 0, "yes": 1, "Yes": 1}
+    sex_values: dict[str, int] = {"Female": 0, "Male": 1}
+    variable_types = get_variable_types(df)
+    encoding = {variable: yes_no for variable in variable_types["binary"][1:]}
+    encoding["Sex"] = sex_values
+    df = df.replace(encoding, inplace=False)
+
+    #Symbolic
+    general_health_encoding = {'Very good':3, 'Excellent':4, 'Fair':1, 'Poor':0, 'Good':1, 'nan':np.nan}
```

Comment:
Should good and fair be the same value or should good be 2?
poor --> fair --> good --> Very good --> Excellent ?

Reply:
Agree